pkg/controller/node/node_controller.go

```diff
@@ -170,7 +170,13 @@ func (ctrl *Controller) checkMasterNodesOnAdd(obj interface{}) {
 		glog.V(4).Infof("We don't care about CRs other than cluster created for scheduler config")
 		return
 	}
-	areMastersSchedulable := scheduler.Spec.MastersSchedulable
+	var areMastersSchedulable bool
+	if scheduler.Spec == (configv1.SchedulerSpec{}) {
+		glog.V(4).Infof("Scheduler spec is nil, so set master as unschedulable")
```

Comment:
spec isn't really nil - why don't we change the `MastersSchedulable` within spec to be a `*bool` while we're still in time to do that?

Reply:
Ya, technically, it's empty struct, I am checking with. I see your point but I think having `bool` is fine too, isn't it?

Reply:
well, yeah, it's ok but I think having a pointer to differentiate set/unset (as this case) would be cleaner, no?

Reply:
>  but I think having a pointer to differentiate set/unset (as this case) would be cleaner, no?

True but we're comparing Spec here which is a struct, not a pointer to struct and the `*bool` is for `MastersSchedulable` field which we added recently. For primitive types like `bool`, I think it's ok not to have pointers and considering we have 3 more days left before feature freeze